Dockerfile_builder

```diff
@@ -0,0 +1,22 @@
+# Dockerfile used to build the policy server binaries
+#
+# This dockerfile could be merged with Dockerfile_musl if Github action get
+# access again to the musl.cc URLS.
+# See more at: https://github.com/orgs/community/discussions/27906
+#
+# TODO if this change got approved. The following container image should be
+# under Kubewarden organizaiton
+FROM ghcr.io/jvanz/policy-server-builder:latest
+ARG TARGETARCH=x86_64
+
+RUN $HOME/.cargo/bin/rustup update
+RUN $HOME/.cargo/bin/rustup target add "${TARGETARCH}-unknown-linux-musl"
+
+COPY . /policy-server
+WORKDIR /policy-server
+
+RUN CC="/${TARGETARCH}-linux-musl-cross/bin/${TARGETARCH}-linux-musl-cc" \
+        LINKER="/${TARGETARCH}-linux-musl-cross/bin/${TARGETARCH}-linux-musl-ld" \
+	PATH="$PATH:/${TARGETARCH}-linux-musl-cross/bin" \
+	$HOME/.cargo/bin/cargo build --target="${TARGETARCH}-unknown-linux-musl" --release
+RUN mv "target/${TARGETARCH}-unknown-linux-musl/release/policy-server" policy-server
```

Comment:
This whole file can be merge with the old `Dockerfile` to remove one container image from the scenario. It will required some changes in the github workflow.